binderhub/tests/test_build.py

```diff
@@ -17,17 +17,26 @@
 
 @pytest.mark.asyncio(timeout=900)
 @pytest.mark.parametrize("slug", [
-    "gh/binderhub-ci-repos/requirements/d687a7f9e6946ab01ef2baa7bd6d5b73c6e904fd",
-    "git/{}/d687a7f9e6946ab01ef2baa7bd6d5b73c6e904fd".format(
+    "gh/binderhub-ci-repos/requirements/20c4fe55a9b2c5011d228545e821b1c7b1723652",
+    "git/{}/20c4fe55a9b2c5011d228545e821b1c7b1723652".format(
```

Comment:
Should we use HEAD here to avoid manually changing the ref for the tests? That also tests the default behavior.

Reply:
When the /master test was added, it was added with the motivation of testing a not already full git hash.

Reply:
I think adding a test case for HEAD is a good idea, especially because that is the default now.

I'd also keep the explicit reference case. My reasoning is that it tests a case that isn't the default and infrequently used/developed for. As a developer you might test master/main/HEAD explicitly but rarely think "ah yes, let's make sure explicit commit hashes still work". This makes it a good task for robots because they don't get bored of testing the same stuff over and over again.

Maybe we can remove the manual work of having to update the commit hash by making the test find out the latest hash and then updating the build slug. Like a templated build slug. However we commit so rarely to the CI repos used here that I'd not spend a lot of time on building this automation and instead would add a comment to let performance improvers from the future know that there is a low hanging fruit here.